pandora-client-web/src/components/header/leaveButton.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { ReactElement, createContext, useCallback, useContext, useState } from 'react';
+import logoutIcon from '../../assets/icons/logout.svg';
+import { HeaderButton } from './HeaderButton';
+import { useShardConnectionInfo } from '../gameContext/shardConnectorContextProvider';
+import { usePlayer, usePlayerData, usePlayerState } from '../gameContext/playerContextProvider';
+import { useCurrentAccount, useDirectoryConnector } from '../gameContext/directoryConnectorContextProvider';
+import { EMPTY, GetLogger, IChatRoomFullInfo } from 'pandora-common';
+import { Button } from '../common/button/button';
+import { useLogout } from '../../networking/account_manager';
+import { useCharacterRestrictionsManager, useChatRoomInfo } from '../gameContext/chatRoomContextProvider';
+import { PlayerCharacter } from '../../character/player';
+import { toast } from 'react-toastify';
+import { TOAST_OPTIONS_ERROR } from '../../persistentToast';
+import { ModalDialog } from '../dialog/dialog';
+import { Column } from '../common/container/container';
+import './leaveButton.scss';
+import { HoverElement } from '../hoverElement/hoverElement';
+
+const leaveButtonContext = createContext(() => { /** noop */ });
+
+export function LeaveButton() {
+	const [show, setShow] = useState(false);
+	const closeDialog = useCallback(() => setShow(false), []);
+
+	return (
+		<leaveButtonContext.Provider value={ closeDialog }>
+			<HeaderButton icon={ logoutIcon } iconAlt='Leave' onClick={ () => setShow(true) } title='Leave' />
+			{
+				show ? <DialogLeave /> : null
+			}
+		</leaveButtonContext.Provider>
+	);
+}
+
+function DialogLeave(): ReactElement {
+	const closeDialog = useContext(leaveButtonContext);
```

Comment:
Nit: Wouldn't it be much simpler to just pass it as a prop instead of context?

Reply:
I didn't want to pass that action all the way down so I felt a context result in a simpler code